p2p/src/protocol.rs

```diff
@@ -750,8 +778,16 @@ impl<T: BlockTree, F: Filters, P: peer::Store> Protocol<T, F, P> {
             NetworkMessage::GetAddr => {
                 self.addrmgr.received_getaddr(&addr);
             }
-            NetworkMessage::GetData(inv) => {
-                (*self.hooks.on_getdata)(addr, inv, &self.upstream);
+            NetworkMessage::GetData(invs) => {
+                let txs = self.txnmgr.get_transaction_hashes();
+
+                for tx in txs {
+                    if invs.contains(&Inventory::Transaction(tx)) {
+                        self.txnmgr.send_transaction(&tx, addr, &self.upstream);
+                    }
+                }
```

Comment:
This should be flipped around: iterate over the invs, and for all invs concerns txs, check if the tx exists.